src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -184,9 +188,65 @@ public static async Task Main(string[] args)
                 {
                     WriteDiagnosticResults(analysisResult.SelectMany(pair => pair.Value.GetAllDiagnostics().Select(j => Tuple.Create(pair.Key, j))).ToImmutableArray(), options.LogFileName);
                 }
+
+                if (options.FormatCode)
+                {
+                    await FormatFilesInSolutionAsync(solution, cancellationToken).ConfigureAwait(false);
+                }
+            }
+        }
+
+        private static async Task FormatFilesInSolutionAsync(Solution solution, CancellationToken cancellationToken)
+        {
+            foreach (var projectId in solution.ProjectIds)
+            {
+                var project = solution.GetProject(projectId);
+                if (project.Language != LanguageNames.CSharp && project.Language != LanguageNames.VisualBasic)
+                {
+                    continue;
+                }
+
+                var isCommentTrivia = project.Language == LanguageNames.CSharp
+                    ? (Func<SyntaxTrivia, bool>)IsCSharpCommentTrivia
+                    : IsVBCommentTrivia;
+
+                foreach (var documentId in project.DocumentIds)
+                {
+                    var document = project.GetDocument(documentId);
+                    if (!document.SupportsSyntaxTree)
+                    {
+                        continue;
+                    }
+
+                    var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    if (GeneratedCodeUtilities.IsGeneratedCode(syntaxTree, isCommentTrivia, cancellationToken))
+                    {
+                        continue;
+                    }
+
+                    var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                    var formattedDocument = await Formatter.FormatAsync(document, documentOptions, cancellationToken).ConfigureAwait(false);
+
+                    var documentText = await formattedDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                    File.WriteAllText(formattedDocument.FilePath, documentText.ToString());
```

Comment:
@mavasani Is there a way to write the files back to disk by using a workspace operation?

Reply:
I updated to use `solution.WithDocumentText(document.Id, documentText)` and `solution.WithDocumentText(document.Id, documentText)`